src/resolve/resolveScalar.js

```diff
@@ -1,10 +1,16 @@
 import { Scalar } from '../ast/Scalar.js'
 
+// match custom string on null/bool
+function matchCustomTest(str, customTest) {
+  if (!customTest) return false
+  return customTest(str)
+}
+
 // falls back to string on no match
 export function resolveScalar(str, tags, scalarFallback) {
-  for (const { format, test, resolve } of tags) {
+  for (const { format, test, customTest, resolve } of tags) {
     if (test) {
-      const match = str.match(test)
+      const match = str.match(test) || matchCustomTest(str, customTest)
```

Comment:
This still isn't the right place for these checks. Really there's no need to modify `resolveScalar` at all here.

Reply:
Removed.